lib/policy_machine_storage_adapters/active_record/postgresql.rb

```diff
@@ -13,17 +13,32 @@ def self.transitive_closure?(ancestor, descendant)
       end
 
       def self.descendants_of(element_or_scope)
-        recursive_query = join_recursive do |query|
-          query.start_with(parent_id: element_or_scope).connect_by(child_id: :parent_id).nocycle
+        #FIXME: Preloading with to_a seems to be necessary because putting complex sql in start_with can
+        # lead to degenerate performance (noticed in ancestors_of call in accessible_objects)
+        # Ideally, fix the SQL so it's both a single call and performant
+        element_or_scope = [*element_or_scope]
+        case element_or_scope.size
+        when 0
+          PolicyElement.none
+        when 1
+          PolicyElement.where('"policy_elements"."id" IN (SELECT assignments__recursive.child_id FROM (WITH RECURSIVE "assignments__recursive" AS ( SELECT "assignments"."id", "assignments"."child_id", "assignments"."parent_id" FROM "assignments" WHERE "assignments"."parent_id" = ? UNION ALL SELECT "assignments"."id", "assignments"."child_id", "assignments"."parent_id" FROM "assignments" INNER JOIN "assignments__recursive" ON "assignments__recursive"."child_id" = "assignments"."parent_id" ) SELECT "assignments__recursive".* FROM "assignments__recursive") AS "assignments__recursive")', element_or_scope.first.id)
```

Comment:
Yeah, I'm honestly not sure. I'll change it if you/commenters like. Mainly was reluctant since I've got it copypasted four times with minor variations, so that's a lot of vertical space to take up.